R/aggregateData.R

```diff
@@ -67,18 +69,10 @@ aggregateData <- function (dat, subunits, units, aggregatemissings = NULL, renam
   }
 
   # add recode values for err: always recode combinations with err to err
-  am <- cbind(am, err = "err") ;  am <- rbind(am, err = "err")
-
-  # check aggregatemissings against standard codes
-  standard_codes <- c("vc", "mvi", "mnr", "mci", "mbd", "mir", "mbi", "err")
-  am_codes <- unique(c(unlist(dimnames(am)), unlist(am)))
+  if(!all(c(am[nrow(am),],am[,ncol(am)]) %in% "err")) {
+    am <- cbind(am, err = "err") ;  am <- rbind(am, err = "err")
+    }
```

Comment:
A user-specified matrix `am` where `err` occurs in the first line (which should be allowed) like 
```
    err   vc    mvi   mnr   mci   mbd   mir   mbi  
err "err" "err" "err" "err" "err" "err" "err" "err"
vc  "err" "vc"  "mvi" "vc"  "mci" "err" "vc"  "mbi"
mvi "err" "mvi" "mvi" "err" "mci" "err" "err" "err"
mnr "err" "vc"  "err" "mnr" "mci" "err" "mir" "mnr"
mci "err" "mci" "mci" "mci" "mci" "err" "mci" "mci"
mbd "err" "err" "err" "err" "err" "mbd" "err" "err"
mir "err" "vc"  "err" "mir" "mci" "err" "mir" "mir"
mbi "err" "mbi" "err" "mnr" "mci" "err" "mir" "mbi"
```
would cause the loop to add errors even though they already exist.

Reply:
You're right, I will change this.